activerecord/lib/active_record/errors.rb

```diff
@@ -273,12 +273,13 @@ class StrictLoadingViolationError < ActiveRecordError
   #     def create
   #       Book.transaction do
   #         book = Book.new(params[:book])
+  #         # If this fails, the user gets generic 500 behavior.
```

Comment:
In this case we can remove `book = Book.new(params[:book])`

Reply:
cool- i'm not totally clear what you want to keep from 08cebde though

I just put the original code back in, maybe make suggestions there?

Reply:
Ah sorry I wasn't clear 🙂 
The TLDR is I'm suggesting to maybe change it to my code suggestion.

Your example had multiple models (`Stats` and `Author`). This is more realistic example for using a transaction.
where you want to rollback multiple operations. This can make the example clearer as we can comment "we rollback this transaction and pass on the exception", which differs from how `ActiveRecord::Rollback` works.
I don't think we need to make the example more complex with different types of rendering and redirecting.

In the suggested code above, I added your `Stats.create!(type: 'book_purchase')` line (I removed the params because I don't think it adds much to the example).
Adding an extra `create` to the transaction makes it clearer that the `Book.create` also gets rolled back.
To keep it simple I don't think with need to add `Author` as well.

Removing the `!` from `Book.create(params[:book])` makes sure that line doesn't raise.
If that is the first database operation nothing gets rolled back. Letting it raise could be confusing (does it get rolled back or is it never saved?).

I moved the `Stats.create!` below the `Book.create` to make it clear `Book.transaction` can also be rolled back by other classes than `Book`